src/visualizers/VisualizerFactorHistogram.ts

```diff
@@ -0,0 +1,206 @@
+         **/
+        binSize: {
+            value: this.binSize,
+            forceType: 'integer',
+            displayName: 'Bin Size',
+            required: true,
+        },
+
+        /** md
+- First Index: The first index to start getting the factors and forming the 
+               histogram from. If the first index is before the first term
+               of the series then the first term of the series will be used
+         **/
+        firstIndex: {
+            value: '' as string | number,
+            forceType: 'integer',
+            displayName: 'First Index',
+            required: false,
+        },
+
+        /** md
+- Number of Terms: The number of terms in the series after the first term.
+                   If this goes past the last term of the sequence it will
+                   show all terms of the sequence after the first index.
+         **/
+        terms: {
+            value: this.terms,
+            forceType: 'integer',
+            displayName: 'Number of Terms',
+            required: true,
+        },
+    }
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        if (this.params.binSize.value < 1) {
+            status.isValid = false
+            status.errors.push('Bin Size can not be less than 1')
+        }
+
+        return status
+    }
+
+    // Obtain the true first index
+    startIndex(): number {
+        if (
+            typeof this.params.firstIndex.value === 'string'
+            || this.firstIndex < this.seq.first
+        ) {
+            return this.seq.first
+        } else {
+            return this.firstIndex
+        }
+    }
+
+    // Obtain the true number of terms
+    endIndex(): number {
+        return Math.min(this.terms + this.startIndex(), this.seq.last)
+    }
+
+    // Create an array with the number of factor of
+    // each element at the corresponding index of the array
+    factorArray(): number[] {
+        const factorArray = []
+        for (let i = this.startIndex(); i < this.endIndex(); i++) {
+            let counter = 0
+            const factors = this.seq.getFactors(i)
+            if (factors) {
+                for (const [base, power] of factors) {
+                    if (base > 0n) {
+                        counter += Number(power)
+                    } else if (base === 0n) {
+                        counter = 0
+                    }
+                }
+            }
+
+            factorArray[i] = counter
+        }
+        return factorArray
+    }
+
+    // Create an array with the frequency of each number
+    // of factors in the corresponding bins
+    binFactorArray(): number[] {
+        const binFactorArray = []
+        const factorArray = this.factorArray()
+        const largestValue = factorArray.reduce(
+            (a, b) => Math.max(a, b),
+            -Infinity
+        )
+        for (let i = 0; i < largestValue + 1; i++) {
+            binFactorArray.push(0)
```

Comment:
Oops, just saw another point I missed going over my last round of review: doesn't this add too many bins when the binSize is > 1 ?  I think it should be `i <= Math.trunc(largestValue/this.binSize)` -- but you should (a) check that out on some examples, and (b) if that's right, you should have a little utility function so you don't have to duplicate that expression with the same one later; maybe call it something like binOf...

Reply:
Looks good.